### PR TITLE
Add ConditionalPut to cli

### PR DESCRIPTION
**Pull request by thundercw**

### CODE REVIEW

**Comment by tamird** [comment]
I'm doing a bit of cleanup in this code; I'll pull this change into my PR. Thanks!

**Comment by thundercw** [comment]
OK

File: cli/kv.go [hunk 1]
```diff
@@ -143,7 +143,43 @@ func runPut(cmd *cobra.Command, args []string) {
...
+
+func runCPut(cmd *cobra.Command, args []string) {
+	if len(args) != 2 && len(args) != 3 {
+		cmd.Usage()
+		return
+	}
+
+	key := proto.Key(unquoteArg(args[0], true /* disallow system keys */))
+	value := proto.Key(unquoteArg(args[1], false))
+	var expValue proto.Key
```
Comment (tamird) [inline cli/kv.go]: why is this a `proto.Key`?

Comment (tamird) [inline cli/kv.go]: I ended up not pulling this PR into #2160 because of this question; take your time addressing it

Comment (thundercw) [inline cli/kv.go]: There is no special reason. I have changed the type of the key,value and expValue to string. 
PTAL.

**Comment by tamird** [comment]
Thanks!

**Pull request merged**